pyresample/utils/cf.py

```diff
@@ -0,0 +1,497 @@
+     'lambert_conformal_conic',
+     'lambert_cylindrical_equal_area',
+     'latitude_longitude',
+     'mercator',
+     'oblique_mercator',
+     'orthographic',
+     'polar_stereographic',
+     'rotated_latitude_longitude',
+     'sinusoidal',
+     'stereographic',
+     'transverse_mercator',
+     'vertical_perspective')
+
+# dictionnary with the standard_names accepted by CF per projection type
+#   this can be used for reading from and writing to CF files
+_valid_cf_coordinate_standardnames = {}
+# specific name for most grid mappings
+_valid_cf_coordinate_standardnames['default'] = dict()
+_valid_cf_coordinate_standardnames['default']['x'] = ('projection_x_coordinate',)
+_valid_cf_coordinate_standardnames['default']['y'] = ('projection_y_coordinate',)
+# specific name for the latitude_longitude grid mapping
+_valid_cf_coordinate_standardnames['latitude_longitude'] = dict()
+_valid_cf_coordinate_standardnames['latitude_longitude']['x'] = ('longitude',)
+_valid_cf_coordinate_standardnames['latitude_longitude']['y'] = ('latitude',)
+# specific name for the rotated_latitude_longitude grid mapping
+_valid_cf_coordinate_standardnames['rotated_latitude_longitude'] = dict()
+_valid_cf_coordinate_standardnames['rotated_latitude_longitude']['x'] = ('grid_longitude',)
+_valid_cf_coordinate_standardnames['rotated_latitude_longitude']['y'] = ('grid_latitude',)
+# specific name for the geostationary grid mapping (we support two flavors)
+_valid_cf_coordinate_standardnames['geostationary'] = dict()
+_valid_cf_coordinate_standardnames['geostationary']['x'] = (
+    'projection_x_angular_coordinate', 'projection_x_coordinate',)
+_valid_cf_coordinate_standardnames['geostationary']['y'] = (
+    'projection_y_angular_coordinate', 'projection_y_coordinate',)
+
+
+def _convert_XY_CF_to_Proj(crs, axis_info):
+    """ In few cases (at present only geostationary) the XY stored in CF files are
+        not directly what Proj (thus pyresample) expect. We need a conversion """
+
+    crs_dict = crs.to_dict()
+    if crs_dict['proj'] == 'geos':
+        # for geostationary projection, the values stored as x/y in CF are not directly
+        #  the x/y along the projection axes, but are rather the scanning angles from
+        #  the satellite. We must multiply them by the height of the satellite.
+        satellite_height = crs_dict['h']
+        for k in ('first', 'last', 'spacing'):
+            axis_info[k] *= satellite_height
+        # the unit is now the default (meters)
+        axis_info['units'] = None
+
+    return axis_info
+
+
+def _load_crs_from_cf_gridmapping(nc_handle, grid_mapping_varname):
+    """ use pyproj to parse the content of the grid_mapping variable and initialize a crs object """
+
+    # check the variable exists
+    try:
+        v = nc_handle[grid_mapping_varname]
+    except KeyError:
+        raise ValueError("There is no variable with name {} in the netCDF file".format(grid_mapping_varname))
```

Comment:
Done. consolidated the Exceptions so that KeyError is sent when a variable doesn't exist, and ValueError in other error cases. https://github.com/TomLav/pyresample/commit/4fc781a896bfc120fafcd53a24bd86ecf583293b